lib/rack/allowed_hosts.rb

```diff
@@ -27,14 +29,20 @@ def allow(*hosts)
     end
 
     def call(env)
+      http_host = env['HTTP_HOST']
+
+      unless http_host.nil?
```

Comment:
In this case, if the `env['HTTP_HOST']` is `nil` we will always return a 403, now, since `nil` won't match one of the allowed values. I think this seems fine, what do you think @jeffjo ?